### ISSUE TITLE
tracker: Fedora 38 changes considerations

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1357]
*Output generated by and [stored alongside](https://pagure.io/fork/dustymabe/fedora-pgm/pgm_scripts/blob/dusty-fcos-changes/f/changes/FCOS-changes.md.orig) ([with modifications](https://pagure.io/fork/dustymabe/fedora-pgm/pgm_scripts/blob/dusty-fcos-changes/f/changes/FCOS-changes.md)) [this](https://pagure.io/fork/dustymabe/fedora-pgm/pgm_scripts/blob/dusty-fcos-changes/f/changes/genFCOSChanges.sh) script in a fork of the [pgm_scripts repo](https://pagure.io/fedora-pgm/pgm_scripts).*

---

**Fedora 38 Accepted System-Wide Changes** ([wiki source](https://fedoraproject.org/wiki/Releases/38/ChangeSet#Fedora_Linux_38_Accepted_System-Wide_Changes))

101. ~~[DNF/RPM Copy on Write enablement for all variants](https://fedoraproject.org/wiki/Changes/RPMCoW)~~
     - ~~RPM Copy on Write provides a better experience for Fedora Users as it reduces the amount of I/O and offsets CPU cost of package decompression. RPM Copy on Write uses reflinking capabilities in btrfs, which is the default filesystem starting from Fedora 33 for most variants. Note that this behavior is not being turned on by default for this Change.~~
     - ~~Tracking bug: [#1915976](https://bugzilla.redhat.com/show_bug.cgi?id=1915976)~~
     - **NOTES:** **JL:** this path of librpm is not used by rpm-ostree. the whole download and unpack path is ostree native and has different tradeoffs. good to keep track of this conceptually, but nothing for FCOS to do here.
     - **NOTES:** **DWM:** this was dropped from consideration for F38
101. ~~[Drop NIS(+) support from PAM](https://fedoraproject.org/wiki/Changes/drop_NIS_support_from_PAM)~~
     - ~~This change is about dropping user-authentication using NIS(+) from PAM.~~
     - ~~Tracking bug: [#2021660](https://bugzilla.redhat.com/show_bug.cgi?id=2021660)~~
     - **NOTES**: **JL**: may affect users who use NIS+? likely not though. If so, we should direct them to e.g. LDAP or FreeIPA as the Change proposal suggests. so overall, skip.
101. ~~[Retire the NIS(+) user-space utility programs](https://fedoraproject.org/wiki/Changes/retire_NIS_user_space_utils)~~
     - ~~This change is about retiring the ypbind, yp-tools, and ypserv packages, and removal of the {nis,yp}domainname user-space utility programs from the hostname package.~~
     - ~~Tracking bug: [#2022386](https://bugzilla.redhat.com/show_bug.cgi?id=2022386)~~
     - **NOTES**: **JL**: may affect users who use NIS+? likely not though. If so, we should direct them to e.g. LDAP or FreeIPA as the Change proposal suggests. so overall, skip.
101. ~~[Changes/LegacyXorgDriverRemoval](https://fedoraproject.org/wiki/Changes/LegacyXorgDriverRemoval)~~
     - ~~This change removes the xorg-x11-drv-vesa and xorg-x11-drv-fbdev driver packages, and associated support code from the xorg-x11-server-Xorg package.~~
     - ~~Tracking bug: [#2078921](https://bugzilla.redhat.com/show_bug.cgi?id=2078921)~~
     - **NOTES**: **DM**: FCOS doesn't ship a graphical desktop. This shouldn't affect us.
101. ~~[Build all JDKs in Fedora against in-tree libraries and with static stdc++lib](https://fedoraproject.org/wiki/Changes/JdkInTreeLibsAndStdclibStatic)~~
     - ~~This is initial step to move JDKs to be more like other JDKs, to build proper transferable images, and to lower certification burden of each binary. Long storyshort, first step in: https://fedoraproject.org/wiki/MoveFedoraJDKsToBecomePortableJDKs~~
     - ~~Tracking bug: [#2097516](https://bugzilla.redhat.com/show_bug.cgi?id=2097516)~~
     - **NOTES**: **DM**: We don't ship java
101. ✔️ [RPM Macros for Build Flags](https://fedoraproject.org/wiki/Changes/RPMMacrosForBuildFlags)
     - Create "extra flag" macros to make it easier for packages to add to the default list of compiler flags.
     - Tracking bug: [#2100610](https://bugzilla.redhat.com/show_bug.cgi?id=2100610)
     - **NOTES**: **DM**: We may want to update our packages, but no need to update anything now.
101. ✔️ [MAC Address Policy none](https://fedoraproject.org/wiki/Changes/MAC_Address_Policy_none)
     - The systemd-udev package installs "/usr/lib/systemd/network/99-default.link", which sets Link.MACAddressPolicy=persistent for all software NIC devices. This proposal is to add to the policy so that we use Link.MACAddressPolicy=none for bond/bridge/team devices.
     - Tracking bug: [#2107754](https://bugzilla.redhat.com/show_bug.cgi?id=2107754)
     - **NOTES**: **DM**: This is our change proposal. See https://github.com/coreos/fedora-coreos-tracker/issues/919
101. ~~[KTLS implementation for GnuTLS](https://fedoraproject.org/wiki/Changes/KTLSSupportForGnuTLS)~~
     - ~~Acceleration of GnuTLS with software Kernel TLS (KTLS)~~
     - ~~Tracking bug: [#2130000](https://bugzilla.redhat.com/show_bug.cgi?id=2130000)~~
     - **NOTES**: **CW**: Anything important we use doesn't use gnutls. Should be nothing for us to do here.
     - **NOTES:** **DWM:** this was dropped from consideration for F38
101. ✔️ [Rpm Sequoia](https://fedoraproject.org/wiki/Changes/RpmSequoia)
     - Change RPM to use Sequoia based OpenPGP parser instead of it's own, flawed and limited implementation.
     - Tracking bug: [#2130122](https://bugzilla.redhat.com/show_bug.cgi?id=2130122)
     - **NOTES**: **JL**: We'll probably use this via libdnf, but it's an implementation detail of RPM and hopefully we have nothing to do here.
101. ~~[Modernize Live Media](https://fedoraproject.org/wiki/Changes/ModernizeLiveMedia)~~
     - ~~Modernize the live media by switching to the "new" live environment setup scripts provided by livesys-scripts and leverage new functionality in dracut to enable support for automatically enabling persistent overlays when flashed to USB sticks.~~
     - ~~Tracking bug: [#2139918](https://bugzilla.redhat.com/show_bug.cgi?id=2139918)~~
     - **NOTES**: **JL**: This is interesting but not relevant since we ship our own live media.
101. ✔️ [Ruby 3.2](https://fedoraproject.org/wiki/Changes/Ruby_3.2)
     - Ruby 3.2 is the latest stable version of Ruby. Many new features and improvements are included for the increasingly diverse and expanding demands for Ruby. With this major update from Ruby 3.1 in Fedora 37 to Ruby 3.2 in Fedora 38, Fedora becomes the superior Ruby development platform.
     - Tracking bug: [#2144054](https://bugzilla.redhat.com/show_bug.cgi?id=2144054)
     - **NOTES**: **TR**: We don't ship ruby. Nothing to do here.
101. ✔️ [Reproducible builds: Clamp build mtimes to $SOURCE_DATE_EPOCH](https://fedoraproject.org/wiki/Changes/ReproducibleBuildsClampMtimes)
     - The %clamp_mtime_to_source_date_epoch RPM macro will be set to 1. When an RPM package is built, mtimes of packaged files will be clamped to $SOURCE_DATE_EPOCH which is already set to the date of the latest %changelog entry. As a result, more RPM packages will be reproducible: The actual modification time of files that are e.g. modified in the %prep section or built in the %build section will not be reflected in the resulting RPM packages. Files in RPM packages will have mtimes that are independent of the time of the actual build.
     - Tracking bug: [#2149310](https://bugzilla.redhat.com/show_bug.cgi?id=2149310)
     - **NOTES**: **JL**: This is about rpm build improvements and no explicit action needed from us. It might have an indirect impact via change detection (in a good way).
101. ~~[Ostree Native Container (Phase 2, stable)](https://fedoraproject.org/wiki/Changes/OstreeNativeContainerStable)~~
     - ~~Continue the work done in https://fedoraproject.org/wiki/Changes/OstreeNativeContainer but in an officially stable format, and expanded to cover more OSTree-based editions.  This goes "all in" on being container-native and significantly changes the technology and user emphasis.~~
     - ~~Tracking bug: [#2151321](https://bugzilla.redhat.com/show_bug.cgi?id=2151321)~~
     - **NOTES**: **DM**: This is our proposal. See https://github.com/coreos/fedora-coreos-tracker/issues/1363
101. ✔️ [LLVM 16](https://fedoraproject.org/wiki/Changes/LLVM-16)
     - Update all llvm sub-projects in Fedora Linux to version 16.
     - Tracking bug: [#2152199](https://bugzilla.redhat.com/show_bug.cgi?id=2152199)
     - **NOTES**: **DM**: This should not affect us or should be transparent.
101. ~~[Add Fedora Auto Firstboot Services to desktop variants](https://fedoraproject.org/wiki/Changes/AutoFirstBootServices)~~
     - ~~Add fedora-autofirstboot to desktop variants to run a predetermined set of tasks on first boot after post installation, notably installing codecs and cleaning up installer packages from the installed system.~~
     - ~~Tracking bug: [#2152200](https://bugzilla.redhat.com/show_bug.cgi?id=2152200)~~
     - **NOTES**: **DM**: This is targetted at desktop variants, but we should make sure it doesn't creep in somehow.
     - **NOTES:** **DWM:** this was dropped from consideration for F38
101. ✔️ [Golang 1.20](https://fedoraproject.org/wiki/Changes/golang1.20)
     - Update of Go (golang package) to the upcoming version 1.20 in Fedora 38.
     - Tracking bug: [#2155076](https://bugzilla.redhat.com/show_bug.cgi?id=2155076)
     - **NOTES**: **DM**: for golang 1.20 we don't foresee any problems, but we should pro-actively test before hand to make sure there are no surprises
101. ✔️ [GNU Make version 4.4](https://fedoraproject.org/wiki/Changes/MAKE44)
     - Rebase GNU make in Fedora 38 from make version 4.3 to make version 4.4.
     - Tracking bug: [#2155276](https://bugzilla.redhat.com/show_bug.cgi?id=2155276)
     - **NOTES**: **JL**: We think this should be a transparent update for us.
101. ⚠️ [Restore stricter SSH hostkeys permissions](https://fedoraproject.org/wiki/Changes/SSHKeySignSuidBit)
     - We want to - drop a downstream-only patch to ssh permitting group-readable ssh host keys - drop a ssh_keys group - restore suid bit instead of sgid on a helper utility ssh-keysign
     - Tracking bug: [#2158230](https://bugzilla.redhat.com/show_bug.cgi?id=2158230)
     - **NOTES**: **DM**: We have dealt with some fallout from this in `rawhide` already (https://github.com/coreos/fedora-coreos-tracker/issues/1391, https://github.com/coreos/fedora-coreos-tracker/issues/1394) and are [working with upstream](https://src.fedoraproject.org/rpms/openssh/pull-request/39) on some more elegant solutions.
101. ✔️ [Add _FORTIFY_SOURCE=3 to distribution build flags](https://fedoraproject.org/wiki/Changes/Add_FORTIFY_SOURCE=3_to_distribution_build_flags)
     - Replace the current _FORTIFY_SOURCE=2 with _FORTIFY_SOURCE=3 to improve mitigation of security issues arising from buffer overflows in packages in Fedora.
     - Tracking bug: [#2158232](https://bugzilla.redhat.com/show_bug.cgi?id=2158232)
     - **NOTES**: **DM**: This should be transparent to us. If any problems arise it should happen at rpm build time.
101. ✔️ [Perl: Replace versioned MODULE_COMPAT_ requires by RPM dependency generator](https://fedoraproject.org/wiki/Changes/Perl_replace_MODULE_COMPAT_by_generator)
     - A perl(:MODULE_COMPAT_%(eval "`%{__perl} -V:version`"; echo $version)) run-time dependency will be removed from all Perl spec files. The replacement will be generated by dependency generator which will be placed in perl-generators.
     - Tracking bug: [#2158233](https://bugzilla.redhat.com/show_bug.cgi?id=2158233)
     - **NOTES**: **DM**: We don't ship perl.
101. ✔️ [Add -fno-omit-frame-pointer and -mno-omit-leaf-frame-pointer to default compilation flags](https://fedoraproject.org/wiki/Changes/fno-omit-frame-pointer)
     - Fedora will add -fno-omit-frame-pointer and -mno-omit-leaf-frame-pointer to the default C/C++ compilation flags, which will improve the effectiveness of profiling and debugging tools.
     - Tracking bug: [#2158246](https://bugzilla.redhat.com/show_bug.cgi?id=2158246)
     - **NOTES**: **JL**: should be transparent to us, no changes for us to make.
101. ✔️ [Noto Fonts For Two Languages](https://fedoraproject.org/wiki/Changes/NotoFontsForTwoLang)
     - Changes the default font for two languages to Noto Fonts.
     - Tracking bug: [#2161751](https://bugzilla.redhat.com/show_bug.cgi?id=2161751)
     - **NOTES**: **JL**: We don't ship fonts. This won't apply to us.
101. ⚠️ [Shorter Shutdown Timer](https://fedoraproject.org/wiki/Changes/Shorter_Shutdown_Timer)
     - A downstream configuration change to reduce the systemd unit timeout from 2 minutes to 45 seconds and send SIGABRT to generate a core dump before SIGKILL.
     - Tracking bug: [#2161753](https://bugzilla.redhat.com/show_bug.cgi?id=2161753)
     - **NOTES**: **DM**: We will need to consider the shutdown timeout change and if we want to accept it or add our own configuration override: https://github.com/coreos/fedora-coreos-tracker/issues/1404
101. ✔️ [GNU Toolchain Update (gcc 13.0, binutils 2.39, glibc 2.37, gdb 12.1)](https://fedoraproject.org/wiki/Changes/GNUToolchainF38)
     - Update the Fedora 38 GNU Toolchain to gcc 13.0, binutils 2.39, and glibc 2.37.
     - Tracking bug: [#2161755](https://bugzilla.redhat.com/show_bug.cgi?id=2161755)
     - **NOTES**: **DM**: This change should be picked up by us transparently; no active changes for us to make.
101. ✔️ [Rpmautospec by Default](https://fedoraproject.org/wiki/Changes/Rpmautospec_by_Default)
     - Rpmautospec (%autorelease and %autochangelog) is recommended as the default approach. Packaging Guidelines and other documentation are adjusted to describe this approach first. Various tools that provide spec file templates are adjusted.
     - Tracking bug: [#2161756](https://bugzilla.redhat.com/show_bug.cgi?id=2161756)
     - **NOTES**: **DM**: Nothing for FCOS to do specifically but the maintainers of the packages we own may choose to update.
101. ✔️ [Unfiltered Flathub](https://fedoraproject.org/wiki/Changes/UnfilteredFlathub)
     - Fedora Workstation's existing third party repo feature allows users to enable a selection of software repos that are hosted by external organizations. This selection has included a filtered version of Flathub since F35, which provides access to a small number of Flathub apps. This change would remove the filtering from our Flathub offering, so that users can enable a complete version of Flathub using the third party repositories feature. In the graphical software manager app, Flathub packages will only be selected by default when no Fedora package is available.
     - Tracking bug: [#2164894](https://bugzilla.redhat.com/show_bug.cgi?id=2164894)
     - **NOTES**: **DM**: We don't ship flatpak. No change for us.
101. ✔️ [z13 as the Baseline for IBM Z Hardware](https://fedoraproject.org/wiki/Changes/z13BaselineForIBMZ)
     - Move the minimal architecture level for IBM Z hardware (the s390x architecture) to z13 to benefit from the new features and better performance in the newer CPU.
     - Tracking bug: [#2115105](https://bugzilla.redhat.com/show_bug.cgi?id=2115105)
     - **NOTES**: **DM**: nothing for us to do as this is just gcc and kernel compiler flags, which we'll pick up transparently.
     Hopefully our users aren't using anything older than z13.
101. ✔️ [Pcre Deprecation](https://fedoraproject.org/wiki/PcreDeprecation)
     - Upstream stopped the support for the old 'pcre' package. It only supports the new 'pcre2' version, so Fedora should deprecate it so it could later be retired and removed from Fedora entirely.
     - Tracking bug: [#2127507](https://bugzilla.redhat.com/show_bug.cgi?id=2127507)
     - **NOTES**: **DM**: the pcre package was removed in f38 from FCOS; nothing further for us to do
101. ✔️ [X Server Prohibits Byte-swapped Clients](https://fedoraproject.org/wiki/Changes/XServerProhibitsByteSwappedClients)
     - X server implementations (e.g. Xorg and Xwayland) will (by default) no longer allow clients with different endianess to connect.
     - Tracking bug: [#2159489](https://bugzilla.redhat.com/show_bug.cgi?id=2159489)
     - **NOTES**: **DM**: this shouldn't affect us, no X servers in FCOS
101. ✔️ [Unified Kernel Support Phase 1](https://fedoraproject.org/wiki/Changes/Unified_Kernel_Support_Phase_1)
     - Add support for unified kernels images to Fedora.
     - Tracking bug: [#2159490](https://bugzilla.redhat.com/show_bug.cgi?id=2159490)
     - **NOTES**: **DM**: no immediate action needed, but part of a larger discussion around confidential computing

---

**Fedora 38 Accepted Self-Contained Changes** ([wiki source](https://fedoraproject.org/wiki/Releases/38/ChangeSet#Fedora_Linux_38_Accepted_Self-Contained_Changes))

201. ~~[glibc 32 Build Adjustments](https://fedoraproject.org/wiki/Changes/glibc32_Build_Adjustments)~~
     - ~~The glibc32 package is a special package used by gcc and a few other packages to work around the lack of RPM multilib repository support in Koji. It is difficult to maintain, and the current approach raises questions regarding (L)GPL compliance.~~
     - ~~Tracking bug: [#1598524](https://bugzilla.redhat.com/show_bug.cgi?id=1598524)~~
     - **NOTES:** **JL:** none of our components use glibc32 AFAIK, so should be good to skip.
201. ~~[Modular GNOME Keyring services](https://fedoraproject.org/wiki/Changes/ModularGnomeKeyring)~~
     - ~~The monolithic daemon provided by GNOME Keyring will be split into dedicated sub-daemons, so that they can be consistently managed by systemd.~~
     - ~~Tracking bug: [#1899998](https://bugzilla.redhat.com/show_bug.cgi?id=1899998)~~
     - **NOTES:** **LB**: we don't ship these GNOME services in FCOS
     - **NOTES:** **DWM:** this was dropped from consideration for F38
201. ✔️ [Wayland by Default for SDDM](https://fedoraproject.org/wiki/Changes/WaylandByDefaultForSDDM)
     - Change the default display server mode for SDDM to use a Wayland-based greeter rather than an X11-based one.
     - Tracking bug: [#2050378](https://bugzilla.redhat.com/show_bug.cgi?id=2050378)
     - **NOTES:** **LB**: we don't ship `sddm` in FCOS
201. ✔️ [Build Fedora IoT Artifacts with osbuild](https://fedoraproject.org/wiki/Changes/IoTArtifactsWithOSBuild)
     - Build the key Fedora IoT artifacts such as the raw images and the traditional anaconda installer with osbuild.
     - Tracking bug: [#2075530](https://bugzilla.redhat.com/show_bug.cgi?id=2075530)
     - **NOTES**: **DM**: This change is specific to Fedora IoT. No action needed for FCOS.
201. ✔️ [Haskell GHC 9.2 & Stackage LTS 20](https://fedoraproject.org/wiki/Changes/Haskell_GHC_9.2_and_Stackage_20)
     - The GHC Haskell compiler will be updated from major version 8.10 to 9.2, and Haskell packages will be updated from Stackage LTS 18 to LTS 20 versions.
     - Tracking bug: [#2080355](https://bugzilla.redhat.com/show_bug.cgi?id=2080355)
     - **NOTES**: **DM**: We don't ship anything that uses haskell.
201. ✔️ [Major upgrade of Microdnf](https://fedoraproject.org/wiki/Changes/MajorUpgradeOfMicrodnf)
     - A major upgrade of Microdnf is the first step in the evolution of package management in Fedora. The new package manager (provided by dnf5 package) has ambitions to provide all major features of DNF without losing its minimal footprint.
     - Tracking bug: [#2080358](https://bugzilla.redhat.com/show_bug.cgi?id=2080358)
     - **NOTES**: **TR**: We don't use microdnf.
201. ✔️ [SPDX License Phase 1](https://fedoraproject.org/wiki/Changes/SPDX_Licenses_Phase_1)
     - Introduce tooling and data allowing package maintainers to transition from Fedora's existing short license names to standardized SPDX license expressions. Update and improve Fedora-legal documentation related to licensing, and move off of wiki.
     - Tracking bug: [#2096410](https://bugzilla.redhat.com/show_bug.cgi?id=2096410)
     - **NOTES**: **JL**: We may need to update our packages to use new License identifiers, but for now it's opt-in.
201. ✔️ [Node.js Repackaging](https://fedoraproject.org/wiki/Changes/NodejsRepackaging)
     - We are reworking the Node.js packaging to make Node.js versions available as parallel-installable packages.
     - Tracking bug: [#2130002](https://bugzilla.redhat.com/show_bug.cgi?id=2130002)
     - **NOTES**: **DM**: Skip. We don't use or include Node.js.
201. ✔️ [PHP 8.2](https://fedoraproject.org/wiki/Changes/php82)
     - Update the PHP stack in Fedora to the latest version 8.2.x
     - Tracking bug: [#2132433](https://bugzilla.redhat.com/show_bug.cgi?id=2132433)
     - **NOTES**: **DM**: Skip. We don't use or include PHP.
201. ✔️ [SWIG 4.1.0](https://fedoraproject.org/wiki/Changes/swig410)
     - Update the SWIG in Fedora to the latest version 4.1.0. New version was released in October 24nd 2022. See RELEASENOTES for more details about new release.
     - Tracking bug: [#2132828](https://bugzilla.redhat.com/show_bug.cgi?id=2132828)
     - **NOTES**: **JL**: Skip. None of our direct projects use SWIG (libdnf does, but we don't enable it for rpm-ostree).
201. ✔️ [Deprecate python-toml](https://fedoraproject.org/wiki/Changes/DeprecatePythonToml)
     - The python-toml (python3-toml) package will be deprecated in Fedora 38. The upstream toml package is considered unmaintained (see description) and Python 3.11 contains a TOML-reading library in the standard library. Existing Fedora packages depend on python-toml, so we cannot remove it yet. Packagers are encouraged to work with upstreams to switch to tomllib/tomli for reading toml or tomli-w for writing it. But python-toml remains available until it is a leaf package, it will be removed then (possibly not yet in Fedora 38).
     - Tracking bug: [#2137375](https://bugzilla.redhat.com/show_bug.cgi?id=2137375)
     - **NOTES**: **DM**: FCOS doesn't use python-toml, but our `config-bot` build tooling does. Opened https://github.com/coreos/fedora-coreos-releng-automation/issues/164
201. ✔️ [PostgreSQL 15](https://fedoraproject.org/wiki/Changes/PostgreSQL_15)
     - Update of PostgreSQL (postgresql and libpq components) in Fedora from version 14 to version 15 in the non-modular (main) builds.
     - Tracking bug: [#2139102](https://bugzilla.redhat.com/show_bug.cgi?id=2139102)
     - **NOTES**: **DM**: We don't include postgresql database in FCOS.
201. ✔️ [LXQt image for aarch64](https://fedoraproject.org/wiki/Changes/LXQt_image_for_aarch64)
     - Generate LXQt image (both iso and disk image) for aarch64 architecture.
     - Tracking bug: [#2139111](https://bugzilla.redhat.com/show_bug.cgi?id=2139111)
     - **NOTES**: **DM**: This is specific to the LXQt desktop.
201. ✔️ [Remove initial-setup from KDE Spin & Kinoite](https://fedoraproject.org/wiki/Changes/KDERemoveInitialSetup)
     - Initial Setup is an application that can run during the first start of a newly installed computer and makes it possible to configure the computer according to the needs of the user. We currently don't use most of the functionality of Initial Setup in the main KDE Spin and Kinoite installation ISOs as everything gets configured at installation time via Anaconda. We thus want to remove this package from the installation ISOs while keeping it where we currently need it (pre-installed disk images, etc.). Note that an "initial setup" app is still needed to enable OEM-style installations (https://askubuntu.com/questions/1386806/what-is-oem-installation-regarding-linux-distributions) of the KDE Spin/Kinoite (like Fedora Workstation/Silverblue) so we're planning on introducing a more KDE native application as a replacement once it is ready, but that may not happen as part of this change.
     - Tracking bug: [#2150045](https://bugzilla.redhat.com/show_bug.cgi?id=2150045)
     - **NOTES**: **DM**: This is specific to KDE* desktops.
201. ✔️ [Prevent from building RPM packages providing python3dist(...) = 0](https://fedoraproject.org/wiki/Changes/Prevent-Providing-python3dist(pkg)=0)
     - It sometimes happens that Python packages succeed to build as RPM with incorrect version metadata. They generate a wrong provide in format python3dist(...) = 0 and python3.Xdist(...) = 0. While version 0 (or equal versions like 0.0 or 0.0.0) is probably technically valid, in most cases this indicates a packaging error.  We propose to prevent this error from happening by explicitly failing the RPM build instead of generating such provides.
     - Tracking bug: [#2150046](https://bugzilla.redhat.com/show_bug.cgi?id=2150046)
     - **NOTES**: **DM**: This looks like it's a build tooling thing and we don't ship python things in FCOS anyway.
201. ✔️ [MobilityPhoshImage](https://fedoraproject.org/wiki/Changes/MobilityPhoshImage)
     - Phosh is a Wayland shell for mobile devices based on Gnome. The mobility SIG has packaged up Phosh and related packages into a 'phosh-desktop' package group and would like to start making x86_64 and aarch64 images for mobile devices.
     - Tracking bug: [#2150052](https://bugzilla.redhat.com/show_bug.cgi?id=2150052)
     - **NOTES**: **DM**: This is a new deliverable from Fedora with no implications for FCOS.
201. ~~[Enable bootupd for Fedora Silverblue & Kinoite](https://fedoraproject.org/wiki/Changes/FedoraSilverblueBootupd)~~
     - ~~By design, ostree does not manage bootloader updates as they can not (yet) happen in a safe fashion. To solve this issue, bootupd (https://github.com/coreos/bootupd) was created. bootupd is a small socket activated program that takes care of updating the bootloader. It currently only supports EFI booted systems and rpm-ostree based systems. The updates are triggered by an administrator and are not (yet) automated for safety reasons. This change is about enabling bootupd integration in Fedora Silverblue and Fedora Kinoite to make bootloader updates easier. bootupd is already used in Fedora CoreOS.~~
     - ~~Tracking bug: [#2150982](https://bugzilla.redhat.com/show_bug.cgi?id=2150982)~~
     - **NOTES**: **JL**: This is Silverblue and Kinoite specific. Nothing to do for FCOS.
201. ~~[Build Fedora Silverblue & Kinoite using rpm-ostree unified core mode](https://fedoraproject.org/wiki/Changes/FedoraSilverblueUnifiedCore)~~
     - ~~rpm-ostree upstream development is focusing on the "unified core" mode and the previous mode is being deprecated. Fedora Silverblue and Fedora Kinoite are currently building using the old mode and we've wanted to move over for a while. The main advantage of the unified core mode is that it is stricter and safer, while enabling some post processing steps to happen during or after the image build.~~
     - ~~Tracking bug: [#2150984](https://bugzilla.redhat.com/show_bug.cgi?id=2150984)~~
     - **NOTES**: **DM**: This is specific to the build tooling used to create Silverblue/Kinoite and should not affect FCOS.
201. ✔️ [Remove Guile Support from GDB](https://fedoraproject.org/wiki/Changes/RemoveGuileFromGdb)
     - Remove Guile extension language support from the GDB package in favor of the widely tested and feature-rich Python support
     - Tracking bug: [#2151328](https://bugzilla.redhat.com/show_bug.cgi?id=2151328)
     - **NOTES**: **JL**: We don't ship gdb.
201. ✔️ [libpinyin 2.8](https://fedoraproject.org/wiki/Changes/libpinyin_2.8)
     - The libpinyin 2.8 will provide phrase suggestion candidate and longer pinyin candidate features.
     - Tracking bug: [#2155080](https://bugzilla.redhat.com/show_bug.cgi?id=2155080)
     - **NOTES**: **DM**: We don't ship libpinyin.
201. ✔️ [Fedora Sway Spin](https://fedoraproject.org/wiki/Changes/Fedora_Sway_Spin)
     - Create an official Fedora Spin shipping the up-and-coming Sway window manager. Following the footsteps of the recently shipped i3 spin.
     - Tracking bug: [#2155081](https://bugzilla.redhat.com/show_bug.cgi?id=2155081)
     - **NOTES**: **DM**: This is another variant of Fedora.
201. ✔️ [Upgrade ImageMagick to version 7](https://fedoraproject.org/wiki/Changes/ImageMagick7)
     - Upgrade ImageMagick to the latest 7.x version.
     - Tracking bug: [#2158235](https://bugzilla.redhat.com/show_bug.cgi?id=2158235)
     - **NOTES**: **DM**: We don't ship imagemagick.
201. ✔️ [Fedora Budgie Spin](https://fedoraproject.org/wiki/Changes/FedoraBudgie)
     - Creation of an official Fedora Spin shipping the Budgie Desktop environment.
     - Tracking bug: [#2158236](https://bugzilla.redhat.com/show_bug.cgi?id=2158236)
     - **NOTES**: **DM**: This is another variant of Fedora.
201. ✔️ [Use mdadm for BIOS RAID Support in Anaconda](https://fedoraproject.org/wiki/Changes/UseMdadmForBIOSRAIDInAnaconda)
     - Use mdadm instead of dmraid to support BIOS RAID (Firmware RAID or Fake RAID) during the Fedora installation process.
     - Tracking bug: [#2158574](https://bugzilla.redhat.com/show_bug.cgi?id=2158574)
     - **NOTES**: **DM**: This change is specific to Anaconda.
201. ✔️ [Pyramid 2.0](https://fedoraproject.org/wiki/Changes/Pyramid2.0)
     - Update Pyramid (package python-pyramid within Fedora) to latest major version.
     - Tracking bug: [#2159493](https://bugzilla.redhat.com/show_bug.cgi?id=2159493)
     - **NOTES**: **DM**: We don't ship python-pyramid.
201. ✔️ [Xfce-4.18](https://fedoraproject.org/wiki/Changes/xfce-4.18)
     - Xfce 4.18 is a stable release with proven components, provide features to both new and power users alike. This change proposal is submitted to sync fedora packages with the latest upstream release.
     - Tracking bug: [#2159494](https://bugzilla.redhat.com/show_bug.cgi?id=2159494)
     - **NOTES**: **DM**: This is another variant of Fedora.
201. ✔️ [FPC repackaging](https://fedoraproject.org/wiki/Changes/F38-FPC-repackaging)
     - Split the fpc package (the Free Pascal Compiler) into several sub-packages (built from the same spec file).
     - Tracking bug: [#2161763](https://bugzilla.redhat.com/show_bug.cgi?id=2161763)
     - **NOTES**: **JL**: We don't ship this package.
201. ✔️ [TeXLive2022](https://fedoraproject.org/wiki/Changes/TeXLive2022)
     - Update the TeXLive engines and components in Fedora to the 2022 version. This will improve TeX document processing, conversion, and internationalization, which is used by some Fedora packages (and users).
     - Tracking bug: [#2163527](https://bugzilla.redhat.com/show_bug.cgi?id=2163527)
     - **NOTES**: **JL**: We don't ship this package.
201. ✔️ [Noto CJK Variable Fonts](https://fedoraproject.org/wiki/Changes/Noto_CJK_Variable_Fonts)
     - Switch the default Noto CJK fonts for Chinese, Japanese and Korean from static to variable fonts.
     - Tracking bug: [#2164514](https://bugzilla.redhat.com/show_bug.cgi?id=2164514)
     - **NOTES**: **JL**: We don't ship this package.
201. ✔️ [IPP-USB as a weak dependency of CUPS and sane-airscan](https://fedoraproject.org/wiki/Changes/IPPUSBasPrintScanDependency)
     - Add ipp-usb as a weak dependency of packages which provide support for driverless printing (cups), driverless scanning (sane-airscan) and driverless fax for USB devices capable of using driverless functionality (how to find out whether your USB device is driverless here), so such devices will work without a specific driver. ipp-usb design conflicts with the way how drivers work with the device, so a user intervention is required after upgrade.
     - Tracking bug: [#2165612](https://bugzilla.redhat.com/show_bug.cgi?id=2165612)
     - **NOTES**: **JL**: We don't ship this package.
201. ✔️ [cups-filters 2.0b](https://fedoraproject.org/wiki/Changes/Cups-filters2.0b)
     - The cups-filters project has been split into five projects in the new major version 2.0b - cups-filters, libcupsfilters, libppd, braille-printer-app and cups-browsed - the new projects will be packaged and cups-filters package will be rebased to version 2.0b once the other projects are available in rawhide. All projects are now united on Apache Software License 2.0.
     - Tracking bug: [#2166687](https://bugzilla.redhat.com/show_bug.cgi?id=2166687)
     - **NOTES**: **DM**: We don't ship cups.
201. ✔️ [IoT Simplified Installer](https://fedoraproject.org/wiki/Changes/IoTSimplifiedInstaller)
     - Offer Fedora IoT users a new method to create and deploy customized Fedora IoT disk images using a new installer called Simplified Installer.
     - Tracking bug: [#2166689](https://bugzilla.redhat.com/show_bug.cgi?id=2166689)
     - **NOTES**: **DM**: This is specific to Fedora IoT, but that borrows technology from coreos-installer, which we own.
201. ✔️ [Update python-packaging to version >= 22](https://fedoraproject.org/wiki/Changes/Update_python-packaging_to_version_22_plus)
     - Update to a new upstream release of python-packaging contains a breaking change. Since version 22+ upstream removed support for LegacySpecifier and LegacyVersion some packages will break. This is a breaking change and projects are encouraged to use versions adherent to PEP 440.
     - Tracking bug: [#2166942](https://bugzilla.redhat.com/show_bug.cgi?id=2166942)
     - **NOTES**: **DM**: We don't ship python.

**Comment by dustymabe** [comment]
Updated items that were discussed in https://github.com/coreos/fedora-coreos-tracker/issues/1222 during the F37 cycle.

**Comment by dustymabe** [comment]
Updated with the results of the [2022-12-14 community meeting](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/message/L27IDLRFKDNZ4DNJWB3BOLC66ASUQIMY/).

**Comment by travier** [comment]
The next update will include https://fedoraproject.org/wiki/Changes/Shorter_Shutdown_Timer which we should discuss as it will impact us and will likely have to me manually changed if we want to keep the existing timeouts. See https://src.fedoraproject.org/rpms/systemd/pull-request/85.

**Comment by travier** [comment]
For 109 (RPM Sequoia), some users might also encounter https://www.scrye.com/wordpress/nirik/2023/01/31/error-rpmdbnextiterator-skipping-in-fedora-38/ if they use SHA-1 signed layered packages.

**Comment by dustymabe** [comment]
Updated the description with the latest set of proposed changes.

**Comment by dustymabe** [comment]
Updated with the results of the [2023-02-01 community meeting](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/message/QST6EEXBEEHADNZYYJQMPL5IDYXTQKQK/).

**Comment by dustymabe** [comment]
Updated the description with the latest set of proposed changes.

**Comment by dustymabe** [comment]
Updated the description with the latest set of proposed changes.

**Comment by dustymabe** [comment]
Ran the script one more time after I noticed some data loss and fixed that problem. Updated the description.

**Comment by dustymabe** [comment]
Updated with the results of the [2023-03-15 community meeting](https://lists.fedoraproject.org/archives/list/coreos@lists.fedoraproject.org/message/YC72QHUK2DHINP3RAF75JNS4J4EW4UYH/).

**Comment by LorisAlbanese** [comment]
upgrading 37 to 38, remote rsyslog with omrelp break actual running conf , on client:

*.* :omrelp:10.56.20.128:2514

and on server side (running EL8)

input(type="imrelp" address="10.56.20.128" port="2514" maxDataSize="15k" keepAlive="on")

with this message on sysctl rsyslog status:

"librelp error 10008 forwarding to server 10.56.20.128:2514 - suspending  [v8.2210.0-4.fc38 try https://www.rsyslog.com>
action 'action-1-omrelp' suspended (module 'omrelp'), retry 0. There should be messages before this one giving the reason for suspension. [v8.2210.0-4.fc38 try https://www.rsyslog.com/e/2007 ]"

But suggestion cannot drive to a solution.

any suggestion? thank you in advance.

**Comment by travier** [comment]
We're now on Fedora 39 so we can close this one. Please file a new issue if you still have problems.
